File: core/src/main/java/com/eloja/core/entity/Produto.java
package com.eloja.core.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.List;

@Entity
@Data
public class Produto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false)
    private String marca;

    private String cor;

    @Lob
    @Column(nullable = false)
    private String descricao;

    @Column(nullable = false)
    private Double valorDeCompra;

    private Double valorDeVenda;

    @Size(min = 1, max = 3)
    private String tamanho;

    @ManyToMany
    @Column(nullable = false)
    private List<Categoria> categoria;

}

File: frontend-react/src/models/Produto.ts
import Categoria from './Categoria';
import { TamanhoProduto } from './enums/TamanhoProduto';

class Produto {
  id?: number;
  nome: string;
  marca: string;
  cor?: string;
  descricao: string;
  valorDeCompra: number;
  valorDeVenda?: number;
  tamanho?: TamanhoProduto;
  categoria: Categoria;

  constructor(
    nome: string,
    marca: string,
    descricao: string,
    valorDeCompra: number,
    tamanho: TamanhoProduto,
    categoria: Categoria,
    id?: number,
    cor?: string,
    valorDeVenda?: number,
  ) {
    this.id = id;
    this.nome = nome;
    this.marca = marca;
    this.cor = cor;
    this.descricao = descricao;
    this.valorDeCompra = valorDeCompra;
    this.valorDeVenda = valorDeVenda;
    this.tamanho = tamanho;
    this.categoria = categoria;
  }
}

export default Produto;
